### PR TITLE
[gcp] Add tags.yml file and upgrade to package spec 3.0

### PR DESCRIPTION
**Pull request by marc-gr**
## What does this PR do?

Add tags.yml file so that integration's dashboards and saved searches are tagged with "Security Solution" and displayed in the Security Solution UI.

Upgrade to package spec 3.0

Add missing dashboard filters and remove orphan dashboard references.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Related issues

- Relates https://github.com/elastic/integrations/issues/7483

## Screenshots

image

### CODE REVIEW

**Review by ShourieG — APPROVED** [review]
LGTM

File: packages/gcp/manifest.yml [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
     title: logo gcp
     size: 32x32
     type: image/svg+xml
-format_version: 2.7.0
+format_version: 3.0.0
 categories:
   - google_cloud
 conditions:
-  kibana.version: ^8.7.1
+  kibana:
```
Comment (kaiyan-sheng) [inline packages/gcp/manifest.yml:15]: Same thing here: Based on https://github.com/elastic/package-spec/pull/567 kibana.version is equal or greater to 8.10.0?

Comment (marc-gr) [reply packages/gcp/manifest.yml:15]: This changed after some discussion to not make it a requirement https://github.com/elastic/package-spec/pull/577

File: packages/gcp/kibana/dashboard/gcp-1ae960c0-f9f8-11eb-bc38-79936db7c106.json [hunk 2]
```diff
@@ -3060,26 +3060,6 @@
             "id": "metrics-*",
             "name": "controlGroup_70d902f0-3d48-4b91-89ee-2719ab7a1cc5:optionsListDataView",
             "type": "index-pattern"
-        },
-        {
```
Comment (tommyers-elastic) [inline packages/gcp/kibana/dashboard/gcp-1ae960c0-f9f8-11eb-bc38-79936db7c106.json:3064]: i found a few of these dangling references in other packages - do you know what they are/were for?

**Review by tommyers-elastic — APPROVED** [review]
LGTM - thanks !

File: packages/gcp/kibana/dashboard/gcp-1ae960c0-f9f8-11eb-bc38-79936db7c106.json (hunk 2 above)
Comment (marc-gr) [reply packages/gcp/kibana/dashboard/gcp-1ae960c0-f9f8-11eb-bc38-79936db7c106.json:3064]: not really, I think they were meaning to target the autogenerated tags? but don't take my word on that

**Pull request merged**